src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,374 @@
+import * as keycode from 'keycode';
+import * as React from 'react';
+import {SBComponent} from 'stylable-react-component';
+import {Stepper} from '../stepper';
+import styles from './time-picker.st.css';
+import {
+    Ampm, ampmLabels, Format,
+    formatTimeChunk, getCircularValue, is12TimeFormat, isTimeSegment,
+    isTouch, isValidValue, Segment, TimeSegment, to24, toAmpm
+} from './utils';
+
+export interface Props {
+    value?: string;
+    onChange?: (value: string) => void;
+    format?: Format;
+    placeholder?: string;
+    disabled?: boolean;
+    label?: string;
+    name?: string;
+}
+
+export interface State {
+    focus: boolean;
+    hh?: string;
+    mm?: string;
+    format: Format;
+    ampm: Ampm;
+    currentSegment: Segment;
+}
+
+const ampmSwitch = {
+    [Ampm.AM]: Ampm.PM,
+    [Ampm.PM]: Ampm.AM,
+    [Ampm.NONE]: Ampm.NONE
+};
+const segments: Segment[] = ['hh', 'mm', 'ampm'];
+
+function propsValueToSegments(value?: string, format?: Format): {hh?: string, mm?: string, ampm: Ampm} {
```

Comment:
the reason I did not put this into utils, because this needed only for React component, but not for general time logic. I did not put it in component either since it does not use `this`